src/components/ThirdPartyCollectionDetailPage/CollectionContextMenu/CollectionContextMenu.tsx

```diff
@@ -36,8 +36,15 @@ export default class CollectionContextMenu extends React.PureComponent<Props> {
     }
   }
 
+  canDelete() {
+    const { collection } = this.props
+    return !isLocked(collection) && !collection.isPublished
```

Comment:
This logic is perfect, but in order to work for TP collections, I think we need to update the `isPublished` attribute once the items are published. We forgot to add that in the collections reducer.
I believe it'd be something like:
```
case PUBLISH_THIRD_PARTY_ITEMS_SUCCESS: {
      const { collectionId } = action.payload
      return {
        ...state,
        data: {
          ...state.data,
          [collectionId]: {
            ...state.data[collectionId],
            isPublished: true
          }
        }
      }
    }
```
In the `collection/reducer.ts` file.
Without this change, the user could publish some items and the button will still be enabled, because the `isPublished` attribute keeps being `false`.

Reply:
good catch! I kinda brushed over that as I though it'd be dealt by https://github.com/decentraland/builder/issues/1874 but I prefer adding it here 👍